File: auth-application/src/main/java/com/example/auth/application/port/in/VerifyMfaUseCase.java
package com.example.auth.application.port.in;

import com.example.auth.application.security.AuthTokens;
import jakarta.validation.constraints.NotBlank;

public interface VerifyMfaUseCase {

    AuthTokens verify(Command cmd);

    record Command(
            @NotBlank String mfaChallengeToken,
            @NotBlank String code,
            String ipAddress,
            String userAgent,
            String deviceLabel) {
    }
}

File: auth-application/src/main/kotlin/com/example/auth/application/port/in/VerifyMfaUseCase.kt
package com.example.auth.application.port.`in`

import com.example.auth.application.security.AuthTokens
import jakarta.validation.constraints.NotBlank

interface VerifyMfaUseCase {

    fun verify(cmd: Command): AuthTokens

    @JvmRecord
    data class Command(
        @field:NotBlank val mfaChallengeToken: String,
        @field:NotBlank val code: String,
        val ipAddress: String?,
        val userAgent: String?,
        val deviceLabel: String?,
    )
}
